src/main/java/com/softserveinc/ita/rozetka/page_objects/BasketPage.java

```diff
@@ -17,32 +16,40 @@ public OrderPage openOrderPage() {
         return new OrderPage();
     }
 
-    public BasketPage increaseProductCount(int numberOfProduct) {
-        $x(format("((//*[@class='button button_color_white" +
-                " button_size_medium cart-counter__button'])[2])[%d]", numberOfProduct)).click();
+    public BasketPage increaseProductCount(int productNumber) {
+        var productSum = $x(format("(//*[@class='cart-product__price'])[%d]", productNumber)).text();
+        $x(format("((//*[@class='button button_color_white button_size_medium cart-counter__button'])[2])[%d]",
+                productNumber)).click();
+        /* checking var "productSum" in the next statement have been added to ensure updating information in the basket window */
+        $x(format("(//*[@class='cart-product__price'])[%d]", productNumber)).shouldNotHave(text(productSum));
         return this;
     }
 
-    public BasketPage decreaseProductCount(int numberOfProduct) {
-        $x(format("((//*[@class='button button_color_white" +
-                " button_size_medium cart-counter__button'])[1])[%d]", numberOfProduct)).click();
+    public BasketPage decreaseProductCount(int productNumber) throws Exception {
+        if (!isDecreaseProductCountEnabled(productNumber)) {
+            throw new Exception("should not further reduction in the count of product items");
```

Comment:
do not use Exception, it's too general, use something that explains what is happening - IllegalStateException etc
fix exception message

Reply:
fixed